src/ucp/api/ucp.h

```diff
@@ -3367,6 +3399,37 @@ ucs_status_t ucp_stream_recv_request_test(void *request, size_t *length_p);
 void ucp_request_cancel(ucp_worker_h worker, void *request);
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Cancel outstanding communications receive requests matching by
+ *        @a params.
+ *
+ * @param [in]  worker       UCP worker.
+ * @param [in]  params       parameters for matching requests to cancel.
+ *
+ * @return UCS_PTR_STATUS(_ptr) - indicates the status. However, all requests
+ *                                matching @a tag are completed immediately or
+ *                                have been started non blocking cancellation
+ *                                procedure and will be completed in finite time.
+ * @return otherwise            - reserved for future use.
+ *
+ * This routine tries to cancel all outstanding communication receive requests
+ * posted with @ref ucp_tag_recv_nb routine on the same @a worker and
+ * matching . After calling this routine, requests will be in completed or
+ * canceled (but not both) state regardless of the status of the target endpoint
+ * associated with the communication request. If the request is completed
+ * successfully, the @ref ucp_tag_recv_callback_t "receive" completion callback
+ * will be called with the @a status argument of the callback set to
+ * @ref UCS_OK, and in a case it is canceled the @a status argument is set to
+ * @ref UCS_ERR_CANCELED. It is important to note that in order to release
+ * requests back to the library the application is responsible for calling
+ * @ref ucp_request_free".
+ */
+ucs_status_ptr_t
+ucp_worker_tag_recv_cancel_all(ucp_worker_h worker,
```

Comment:
maybe better `ucp_tag_recv_cancel_all`, like other funcs managing requests?

Reply:
this routine works in scope of worker